src/Raven.Client/Util/TcpUtils.cs

```diff
@@ -36,10 +40,14 @@ private static void SetTimeouts(TcpClient client, TimeSpan timeout)
             client.SendTimeout = (int)timeout.TotalMilliseconds;
             client.ReceiveTimeout = (int)timeout.TotalMilliseconds;
         }
-
-        internal static async Task<(TcpClient Client, string Url)> ConnectSocketAsync(
-            TcpConnectionInfo connection,
-            TimeSpan timeout,
+        internal static async Task<(TcpClient Client, Stream Stream, string Url, TcpConnectionHeaderMessage.SupportedFeatures SupportedFeatures)> ConnectSecuredTcpSocketAsReplication(
```

Comment:
Better to create a return object, instead of that complex a tuple

Reply:
Done

Reply:
That is still a tuple, no?